src/stratigraphy/util/language_detection.py

```diff
@@ -0,0 +1,43 @@
+"""This module contains functionalities for language detection of a document."""
+
+import fitz
+from langdetect import detect
+from langdetect.lang_detect_exception import LangDetectException
+
+
+def extract_text_from_document(doc: fitz.Document) -> str:
+    """Extracts and processes text from a document.
+
+    Args:
+        doc (fitz.Document): The document to extract text from.
+
+    Returns:
+        str: The extracted and processed text.
+    """
+    text = ""
+    for page in doc:
+        text += page.get_text()
+    text = text.replace("\n", " ")
+
+    # remove all numbers and special characters from text
+    return "".join(e for e in text if (e.isalnum() or e.isspace()) and not e.isdigit())
```

Comment:
Does this keep accented letter such as öäüéàè that might be important for detecting German/French, or only ASCII?

Reply:
Good question. I'll have to check.

Reply:
öäüéàè and the like are kept.